en/docs/samples/adaptive-auth-with-function-lib-sample.md

```diff
@@ -0,0 +1,148 @@
+
+----
+
+If you have your own application, click the button below.
+
+<a class="samplebtn_a" href="../../guides/adaptive-auth-with-function-lib" target="_blank" rel="nofollow noopener">I have my own application</a>
+
+-----
+
+## Scenario
+
+The instructions below guide you through adding an adaptive authentication function using the function library that specifies authentication based on the user's age. In this example, any user who is underage and below the specified age limit (i.e., under the age of 18 years) is restricted access and prevented from logging in to the application.
+
+----
+## Set up Pickup Dispatch sample
+
+{!fragments/pickup-dispatch-saml.md!}
+
+----
+
+{!fragments/add-function-library.md!}
+
+Add a function library with the following properties.
+    
+- **Function Library Name:** age_based
+- **Description:** getAge of the user 
+- **Function Library Script:** 
+    
+    ```javascript
+    function getAge(birthDate) {
+        var today = new Date();
+        var age = today.getFullYear() - birthDate.getFullYear();
+        var m = today.getMonth() - birthDate.getMonth();
+        if (m < 0 || (m === 0 && today.getDate() < birthDate.getDate())) {
+            age--;
+        }
+        return age;
+    }
+     module.exports.getAge = getAge;
+    ```
+
+----
+
+## Configure claims 
+
+1. On the management console, click **Claims > List**.
+
+2. Click `http://wso2.org/claims`. 
+
+3. Click the **Edit** link corresponding to the **BirthDate** claim.
+
+4. Select the **Supported By Default** checkbox to enable the birth date claim. 
+
+    ![enable-dob-claim](../assets/img/samples/enable-dob-claim.png)
+
+5. Create a user called "Alex" and edit the user profile.
+
+    For instructions, see [Adding Users and Roles](insertlink).
+
+6. Enter a birth date that specifies Alex as under 18 years of age. 
+
+    !!! note
+        Enter the birth date in the following format: `<yyyy-mm-dd>`.
+
+7. Create another user called "Kim" and edit the user profile. Enter a birth date that specifies Kim as over 18 years of age. 
+
+-----
+
+## Configure authentication script
+
+1. Click **Service Providers > List** and click **Edit** on the `saml2-web-app-pickup-dispatch.com` service provider.
+
+2. Expand **Local and Outbound Authentication Configuration** and click **Advanced Configuration**.
+
+3. Under script based adaptive authentication editor, enter the following script.
+
+    ```javascript
+    var ageModule = require('age_based.js');
+    
+    // This script will only allow login to application if the user's age is over configured value
+    // The user will be redirected to an error page if the date of birth is not present or user is below configured value
+    
+    var ageLimit = 18;
+    
+    // Error page to redirect unauthorized users,
+    // can be either an absolute url or relative url to server root, or empty/null
```

Comment:
Can either be an absolute URL or a relative URL  to the server root. The value can be empty or null as well.